org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/snippets/CommentSnippetContext.java

```diff
@@ -36,6 +37,13 @@ public boolean isMatch(ICompletionRequest request, Map<String, String> model) {
 			// completion was triggered inside doctype declaration, ignore the snippets
 			return false;
 		}
+		
+		DOMDocument document = request.getXMLDocument();
+		int offset = request.getOffset();
+		if (document.hasProlog() && offset <= document.getProlog().getStart()){

```

Comment:
I will define in DOMDocument `isInBeforeProlog(int offset)` and I will use it in all *SnippetContext classes

Reply:
sounds reasonable. Did you mean something like this?